java/src/org/openqa/selenium/remote/RemoteNetwork.java

```diff
@@ -30,50 +36,90 @@
 @Beta
 class RemoteNetwork implements Network {
 
+  private static final Logger LOG = Logger.getLogger(RemoteNetwork.class.getName());
+
   private final BiDi biDi;
   private final org.openqa.selenium.bidi.module.Network network;
 
-  private final Map<Long, String> authCallbackIdMap = new HashMap<>();
+  private final Map<Long, AuthDetails> authHandlers = new ConcurrentHashMap<>();
+
+  private final AtomicLong callBackId = new AtomicLong(1);
 
   public RemoteNetwork(WebDriver driver) {
     this.biDi = ((HasBiDi) driver).getBiDi();
     this.network = new org.openqa.selenium.bidi.module.Network(driver);
+
+    interceptAuthTraffic();
+  }
+
+  private void interceptAuthTraffic() {
+    this.network.addIntercept(new AddInterceptParameters(InterceptPhase.AUTH_REQUIRED));
+
+    network.onAuthRequired(
+        responseDetails -> {
+          String requestId = responseDetails.getRequest().getRequestId();
+
+          try {
+            URL url = new URL(responseDetails.getRequest().getUrl());
```

Comment:
Prefer to use `URI` rather than `URL` since `URL.equals` does a domain name lookup. You can create one with `URI.create`

Reply:
Alright. Thank you!